src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingController.java

```diff
@@ -0,0 +1,287 @@
+      members.removeIf(partyMember -> partyMember.getPlayer().equals(player));
+      for(int i = 0; i < members.size(); i++) {
+        PartyMemberItemController controller = uiService.loadFxml("theme/play/teammatchmaking/matchmaking_member_card.fxml");
+        controller.setMember(members.get(i));
+        if (members.size() == 1)
+          partyMemberPane.add(controller.getRoot(), 0, 0, 2, 1);
+        else
+          partyMemberPane.add(controller.getRoot(), i % 2, i / 2);
+      }
+    });
+
+    invitePlayerButton.disableProperty().bind(createBooleanBinding(
+        () -> teamMatchmakingService.getParty().getOwner().getId() != playerService.getCurrentPlayer().map(Player::getId).orElse(-1),
+        teamMatchmakingService.getParty().ownerProperty(),
+        playerService.currentPlayerProperty()
+    ));
+    leavePartyButton.disableProperty().bind(createBooleanBinding(() -> teamMatchmakingService.getParty().getMembers().size() <= 1, teamMatchmakingService.getParty().getMembers()));
+
+    teamMatchmakingService.getParty().getMembers().addListener((InvalidationListener) c -> {
+      refreshingLabel.setVisible(false);
+      selectFactionsBasedOnParty();
+    });
+
+    JavaFxUtil.addListener(teamMatchmakingService.getParty().ownerProperty(), (observable, oldValue, newValue) -> {
+      if (matchmakingChatController != null) {
+        matchmakingChatController.closeChannel();
+      }
+      createChannelTab(String.format("#%s'sParty", newValue.getUsername()));
+    });
+    createChannelTab(String.format("#%s'sParty", teamMatchmakingService.getParty().getOwner().getUsername()));
+
+    fafService.requestMatchmakerInfo();
+  }
+
+  private void initializeDynamicChatPosition() {
+    contentPane.widthProperty().addListener((observable, oldValue, newValue) -> {
+      if ((double) newValue < 1115.0) {
+        GridPane.setColumnIndex(chatTabPane, 0);
+        GridPane.setRowIndex(chatTabPane, 1);
+        GridPane.setColumnSpan(chatTabPane, 2);
+        GridPane.setColumnSpan(scrollPane, 2);
+        column2.setMinWidth(0);
+        row2.setMinHeight(200);
+        chatTabPane.pseudoClassStateChanged(CHAT_AT_BOTTOM_PSEUDO_CLASS, true);
+      } else {
+        GridPane.setColumnIndex(chatTabPane, 1);
+        GridPane.setRowIndex(chatTabPane, 0);
+        GridPane.setColumnSpan(chatTabPane, 1);
+        GridPane.setColumnSpan(scrollPane, 1);
+        column2.setMinWidth(400);
+        row2.setMinHeight(0);
+        chatTabPane.pseudoClassStateChanged(CHAT_AT_BOTTOM_PSEUDO_CLASS, false);
+      }
+    });
+  }
+
+  private void initializeUppercaseText() {
+    matchmakerHeadingLabel.setText(i18n.get("teammatchmaking.playerTitle").toUpperCase());
+    partyHeadingLabel.setText(i18n.get("teammatchmaking.partyTitle").toUpperCase());
+    invitePlayerButton.setText(i18n.get("teammatchmaking.invitePlayer").toUpperCase());
+    leavePartyButton.setText(i18n.get("teammatchmaking.leaveParty").toUpperCase());
```

Comment:
Can this not be done via css? Then it would be completely redundant (handled in FXML)

Reply:
I wish! CSS can do that, but the "special" css for javafx cannot :(

Reply:
Shame